File: api/typescript/src/services/sample.ts
import { Request, Response } from "express";
import { Sample, selectAllSamples, selectSampleByID, insertSample, updateSample, deleteSampleData } from "../models/sample";

export const getAllSamples = async (req: Request, res: Response) => {
  try {
    const result = await selectAllSamples();
    return res.status(200).send(result);
  } catch (e) {
    return res.status(500).send({"message": "ng"});
  }
};

export const getSampleByID = async (req: Request, res: Response) => {
  try {
    const id = req.params.id;
    const result = await selectSampleByID(id);
    return res.status(200).send(result);
  } catch (e) {
    return res.status(500).send({"message": "ng"});
  }
};

export const postSample = async (req: Request, res: Response) => {
  try {
    const id = await insertSample(req.body as Sample)
    return res.status(200).send({ id: id });
  } catch (e) {
    return res.status(500).send({"message": "ng"});
  }
};

export const putSample = async (req: Request, res: Response) => {
  try {
    await updateSample(req.body as Sample);
    return res.status(200).send({ message: "ok" });
  } catch (e) {
    return res.status(500).send({ message: "ng"});
  }
};

export const deleteSample = async (req: Request, res: Response) => {
  try {
    const id = req.params.id;
    await deleteSampleData(id);
    return res.status(200).send({ message: "ok"});
  } catch (e) {
    return res.status(500).send({message: "ng"});
  }
};

File: api/rust/app/src/services/sample.rs
use rocket::serde::json::Json;

use crate::models::sample::*;

#[get("/samples")]
pub fn get_samples() -> Json<Vec<Sample>> {
    let vec = select_all_samples();
    Json(vec)
}

#[get("/sample/<id>")]
pub fn get_sample_by_id(id: &str) -> Json<Sample> {
    let value = select_sample_by_id(id);
    Json(value)
}

#[post("/sample", format = "application/json", data = "<sample>")]
pub fn post_sample(sample: Json<Sample>) -> Json<PostResponse> {
    let res = insert_sample(sample.0);
    Json(res)
}

#[put("/sample", format = "application/json", data = "<sample>")]
pub fn put_sample(sample: Json<Sample>) -> Json<Message> {
    let res = update_sample(sample.0);
    Json(res)
}

#[delete("/sample/<id>")]
pub fn delete_sample(id: &str) -> Json<Message> {
    let res = delete_sample_data(id);
    Json(res)
}